includes/woopay/class-woopay-session.php

```diff
@@ -556,6 +532,54 @@ public static function ajax_get_woopay_session() {
 		wp_send_json( self::get_frontend_init_session_request() );
 	}
 
+	/**
+	 * Used to initialize woopay session on frontend
+	 *
+	 * @return void
+	 */
+	public static function ajax_get_woopay_minimum_session_data() {
+		$is_nonce_valid = check_ajax_referer( 'woopay_session_nonce', false, false );
+
+		if ( ! $is_nonce_valid ) {
+			wp_send_json_error(
+				__( 'You aren’t authorized to do that.', 'woocommerce-payments' ),
+				403
+			);
+		}
+
+		$blog_id = Jetpack_Options::get_option('id');
+		if ( empty( $blog_id ) ) {
+			wp_send_json_error(
+				__( 'Could not determine the blog ID.', 'woocommerce-payments' ),
+				503
+			);
+		}
+
+		wp_send_json( self::get_woopay_minimum_session_data() );
+	}
+
+	/**
+	 * Return WooPay minimum session data.
+	 * 
+	 * @return array Array of minimum session data used by WooPay or false on failures.
+	 */
+	public static function get_woopay_minimum_session_data() {
+		$blog_id = Jetpack_Options::get_option('id');
+		if ( empty( $blog_id ) ) {
+			return [];
+		}
+
+		$data = [
+			'blog_id'           => $blog_id,
+			'blog_rest_url'     => get_rest_url(),
+			'blog_checkout_url' => wc_get_checkout_url(),
+			'session_nonce'     => self::create_woopay_nonce( get_current_user_id() ),
+			'store_api_token'   => self::init_store_api_token(),
+		];
+
+		return WooPay_Utilities::encrypt_and_sign_data( $data );
+	}
```

Comment:
To prevent issues related to `openssl` not being enabled on the host, I guess we should add a `WC_Payments_Features::is_client_secret_encryption_eligible()` check before `encrypt_and_sign_data`, similar to how it's done in `get_frontend_init_session_request`.

Reply:
Great catch, I've copied the same check here now in https://github.com/Automattic/woocommerce-payments/pull/8268/commits/4b1b7d4c320e178232d3c1ebea1cfed829d7c355.